data/transactions/stateproof_test.go

```diff
@@ -0,0 +1,148 @@
+// Copyright (C) 2019-2025 Algorand, Inc.
+// This file is part of go-algorand
+//
+// go-algorand is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// go-algorand is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with go-algorand.  If not, see <https://www.gnu.org/licenses/>.
+
+package transactions
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/algorand/go-algorand/config"
+	"github.com/algorand/go-algorand/crypto"
+	"github.com/algorand/go-algorand/crypto/stateproof"
+	"github.com/algorand/go-algorand/data/basics"
+	"github.com/algorand/go-algorand/data/stateproofmsg"
+	"github.com/algorand/go-algorand/protocol"
+	"github.com/algorand/go-algorand/test/partitiontest"
+	"github.com/stretchr/testify/require"
+)
+
+type stateproofTxnTestCase struct {
+	expectedError error
+
+	StateProofInterval uint64
+	fee                basics.MicroAlgos
+	note               []byte
+	group              crypto.Digest
+	lease              [32]byte
+	rekeyValue         basics.Address
+	sender             basics.Address
+}
+
+func TestUnsupportedStateProof(t *testing.T) {
```

Comment:
Slightly weird we moved this test since the check is still in `transaction.go`, but not a big deal.

Reply:
meh its fine